gdx-pay-iosrobovm-apple/src/main/java/com/badlogic/gdx/pay/ios/apple/PurchaseManageriOSApple2.java

```diff
@@ -0,0 +1,679 @@
+            transaction.setPurchaseCost((int) Math.round(product.getPrice().doubleValue() * 100));
+            transaction.setPurchaseCostCurrency(product.getPriceFormatStyle().getCurrencyCode());
+        } else {
+            // product information was empty (not loaded or product didn't exist)
+            transaction.setPurchaseText("Purchased: " + productIdentifier);
+            transaction.setPurchaseCost(0);
+            transaction.setPurchaseCostCurrency(null);
+        }
+
+        transaction.setReversalTime(null);  // no refunds for iOS!
+        transaction.setReversalText(null);
+
+        // there is no SKPaymentTransaction.payment.requestData alternative in StoreKit2
+        transaction.setTransactionData(null);
+
+        // NOTE: although deprecated as of iOS 7, "transactionReceipt" is still available as of iOS 9 & hopefully long there after :)
+        String transactionDataSignature;
+        try {
+            NSData transactionReceipt = t.getJsonRepresentation();
+            transactionDataSignature = transactionReceipt.toBase64EncodedString(NSDataBase64EncodingOptions.None);
+        } catch (Throwable e) {
+            log(LOGTYPELOG, "Transaction.jsonRepresentation appears broken", e);
+            transactionDataSignature = null;
+        }
+        transaction.setTransactionDataSignature(transactionDataSignature);
+
+        // return the transaction
+        return transaction;
+    }
+
+    @Override
+    public Information getInformation(String identifier) {
+        if (products != null) {
+            for (Product p : products) {
+                if (p.getId().equals(identifier)) {
+                    if (numberFormatter == null) {
+                        numberFormatter = new NSNumberFormatter();
+                        numberFormatter.setFormatterBehavior(NSNumberFormatterBehavior._10_4);
+                        numberFormatter.setNumberStyle(NSNumberFormatterStyle.Currency);
+                    }
+                    numberFormatter.setLocale(p.getPriceFormatStyle().getLocale());
+                    return Information.newBuilder()
+                            .localName(p.getDisplayName())
+                            .localDescription(p.getProductDescription())
+                            .localPricing(numberFormatter.format(p.getPrice()))
+                            .priceCurrencyCode(p.getPriceFormatStyle().getLocale().getCurrencyCode())
+                            .priceInCents(MathUtils.ceilPositive(p.getPrice().floatValue() * 100))
+                            .priceAsDouble(p.getPrice().doubleValue())
+                            .freeTrialPeriod(convertToFreeTrialPeriod(p))
+                            .build();
+                }
+            }
+        }
+        return Information.UNAVAILABLE;
+    }
+
+    private FreeTrialPeriod convertToFreeTrialPeriod(Product product) {
+        if (!IosVersion.is_11_2_orAbove()) {
```

Comment:
org.robovm.pods.cocoatouch.storekitrvm.Product is available since iOS 15. This if can be dropped

Reply:
yep